hack/usoc22: Update session 03 - Debugging

razvanvirtan opened the pull request:
One quick mention: this PR removes a file from `/content/en/community/hackathons/sessions/` because I think it was introduced by mistake in the repo from the very beginning, being a data file that should be generated by each student on his local system.

Signed-off-by: Razvan Virtan <virtanrazvan@gmail.com>

StefanJum commented on content/en/community/hackathons/sessions/debugging/content/Practical-Work_Support-Files.md:
````diff
@@ -1,25 +1,25 @@
+Session support files are available [in the repository](https://github.com/unikraft/docs).
 If you already cloned the repository, update it and enter the session directory:
 
 ```
````
```suggestion
```bash
```

StefanJum commented on content/en/community/hackathons/sessions/debugging/content/Practical-Work_Support-Files.md:
````diff
@@ -1,25 +1,25 @@
 
 If you haven't cloned the repository yet, clone it and enter the session directory:
 
 ```
````
```suggestion
```bash
```

StefanJum commented on content/en/community/hackathons/sessions/debugging/content/Practical-Work_08.-Give-Us-Feedback.md:
```diff
@@ -1,3 +1,3 @@
 We want to know how to make the next sessions better.
-For this we need your [feedback](https://forms.gle/9EuzgL1n244Mvqfq8).
+For this, we need your [feedback](https://docs.google.com/forms/d/e/1FAIpQLSdpyEpCW0yxJBSDa7Iq95JUCe5YNnRpxOD31ol4ezpKWnqrhA/viewform).
```
This change should not happen here, since it is event-specific, and sessions in the `community/hackathons/sessions` will be added to every event.
The change should be made in the usoc'22 sessions index file.
See [this file](https://github.com/unikraft/docs/pull/96/files#diff-8f5b68fbca8819baba2f15887b55e7bb70cca9c17541b4c6fbe5ab66b39ff844) for reference.

StefanJum commented on content/en/community/hackathons/sessions/debugging/work/04-tracepoints-tutorial/main.c:
```diff
@@ -0,0 +1,31 @@
+UK_TRACEPOINT(start_trace, "%d", int);
+UK_TRACEPOINT(stop_trace, "%d", int);
+
+void start_status(){
```
```suggestion
void start_status()
{
```

StefanJum commented on content/en/community/hackathons/sessions/debugging/work/04-tracepoints-tutorial/main.c:
```diff
@@ -0,0 +1,31 @@
+	printf("Start tracing\n");
+}
+
+void stop_status(){
```
```suggestion
void stop_status()
{
```

StefanJum commented on content/en/community/hackathons/sessions/debugging/content/Tracepoints_Reading-Trace-Data.md:
````diff
@@ -9,15 +9,15 @@ For this purpose, add the following line to your `~/.gdbinit`:
-In order to collect the data, open GDB with the debug image and connect to your Unikraft instance as described in Section [Using GDB](#using-gdb):
+In order to collect the data, open GDB with the debug image and connect to your Unikraft instance as described in Section [Using GDB](community/hackathons/sessions/debugging/#using-gdb):
 
 ```
````
```suggestion
```bash
```

StefanJum reviewed (changes requested):
@razvanvirtan I left a quick round of review regarding just syntax stuff.
I will test the session and come back with thorough feedback soon.

StefanJum reviewed (changes requested):
Coming with a second round of review.
Besides the comments. there are some suggestions on files that you have not modified, so I will give a list of them since I cannot give code comments.

* I would add an intro [here](https://github.com/unikraft/docs/blob/main/content/en/community/hackathons/sessions/debugging/content/Practical-Work_01.-Tutorial.-Use-GDB-in-Unikraft.md?plain=1#L1), basicly telling students that they need to build the `helloworld` application and provide them a link to some intructions for it (could be `app-helloworld` readme file or [`session 01`](http://localhost:1313/community/hackathons/sessions/baby-steps/#01-building-and-running-the-helloworld-application)).
* Is [this](https://github.com/unikraft/docs/blob/main/content/en/community/hackathons/sessions/debugging/content/Practical-Work_02.-Mystery:-Find-the-secret-using-GDB.md?plain=1#L46) necessary? I don't know how you would use it and it surely gives a lot of unneeded information. If we keep it I would at least add a hint on what to look for.

razvanvirtan commented:
@StefanJum I updated the PR based on your last review.
About the `nm` hint, I think the only way you could use it is to grep for `secret`, find an address and see the data structure that contains all the `surname`, `name` etc. fields the application will require. This could be useful if you want to dump all the data in one run, because you can know how many strings you should find.
However, I agree, this is convoluted and could induce more confusion to the participants, so I removed it.

StefanJum commented on content/en/community/hackathons/sessions/debugging/content/Tracepoints_Reading-Trace-Data.md:
```diff
@@ -38,6 +38,6 @@ This can happen when no tracepoint was ever called.
 
 Any saved trace file can be later processed with the `trace.py` script. In our example:
```
I would add the pull path of the `trace.py` script, specifically the fact that the command below is given from the unikraft core directory.

StefanJum reviewed (changes requested):
@razvanvirtan besides the one comment everything looks file.
The one thing left to do is create a quiz and add it to this ps, as discussed [on discord](https://discord.com/channels/762976922531528725/1003970034340732978/1013856758990643270).

StefanJum reviewed (approved):
Looks good now!
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

razvand reviewed (approved):
Thanks, @razvanvirtan , @StefanJum . I did some updates and will approve it.

Reviewed-by: Razvan Deaconescu <razvan.deaconescu@cs.pub.ro>
Approved-by: Razvan Deaconescu <razvan.deaconescu@cs.pub.ro>
